soroban-env-host/benches/common/modelfit.rs

```diff
@@ -1,13 +1,24 @@
+use nalgebra::{self as na, OMatrix, OVector, U1};
+use num_traits::Pow;
+use soroban_env_host::budget::{MeteredCostComponent, ScaledU64};
 use std::collections::HashSet;
 use std::str::FromStr;
 
-use linregress::{FormulaRegressionBuilder, RegressionDataBuilder};
-
 #[derive(Debug, Default, Clone, PartialEq, PartialOrd)]
-pub struct FPCostModel {
-    pub const_param: f64,
-    pub lin_param: f64,
-    pub r_squared: f64,
+pub(crate) struct FPCostModel {
+    const_param: f64,
+    lin_param: f64,
+    r_squared: f64,
+}
+
+impl From<FPCostModel> for MeteredCostComponent {
+    fn from(model: FPCostModel) -> Self {
+        let (const_term, unscaled_lin_term) = model.params_as_u64();
+        MeteredCostComponent {
+            const_term,
+            lin_term: ScaledU64::from_unscaled_u64(unscaled_lin_term),
```

Comment:
Wait, I think this is still maybe not quite right -- you're still losing all the fractional precision in this conversion.

To preserve the fractional precision, you want to scale the f64 as an f64 first, then convert to integer:

```
let fp_scale_factor: f64 = (1 << COST_MODEL_LIN_TERM_SCALE_BITS) as f64;
let fp_scaled_lin_param: f64 = model.lin_param * fp_scale_factor;
lin_term: ScaledU64 = ScaledU64(fp_scaled_lin_param as u64);
```

You'll probably want to encapsulate that logic inside `ScaledU64` though; perhaps a clean way would be to `impl From<f64> for ScaledU64`, then you could have `lin_term: model.lin_param.into()` in this code.

Reply:
Ah yes, you are right.

Reply:
Fixed it, thanks for the catch!
I want to point out (in case you missed it) that there was some rounding logic that had to be threaded in, besides what you suggested.